api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.expressions.partitions.Partition;
+import java.util.Map;
+
+public interface SupportsPartitions extends Table {
+
+  /**
+   * List all partition names of the table
+   *
+   * @return The list of partition names
+   */
+  String[] listPartitionNames();
```

Comment:
should we add a name to listPartition? such as there are many level partitions like year=xx/month=xx/day=xx, we could pass `year=xx` to retrieve specify partitions. refer from spark
```java
    InternalRow[] listPartitionIdentifiers(String[] names, InternalRow ident);
```

Reply:
also spark listPartion returns the schema beside names, I'm not sure whether it's useful